src/contracts/pods/EigenPod.sol

```diff
@@ -454,24 +508,25 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         // Withdrawal credential proofs should only be processed for "INACTIVE" validators
         require(
             validatorInfo.status == VALIDATOR_STATUS.INACTIVE,
-            "EigenPod.verifyCorrectWithdrawalCredentials: Validator must be inactive to prove withdrawal credentials"
+            "EigenPod._verifyWithdrawalCredentials: validator must be inactive to prove withdrawal credentials"
         );
 
-        // Ensure the `validatorFields` we're proving have the correct withdrawal credentials
+        // Validator should not already be in the process of exiting
+        require(
+            validatorFields.getExitEpoch() == FAR_FUTURE_EPOCH,
```

Comment:
do we need this check?

Reply:
i had originally included it because `verifyWithdrawalCredentials` was requiring that `beaconTimestamp >= currentCheckpointTimestamp`. For the `==` case, if your validator is already exited, you may end up proving withdrawal credentials but your validator's ETH is already in the pod, meaning you'd be able to double your shares.

I removed the `==` condition though and made it strictly greater than. MAYBE we don't need this check... but IMO it's probably just safer to keep it? If your validator is about to exit, maybe don't verify its WC?

Reply:
Actually I thought of a reason we still need this check -- if these things happen *within the same epoch, in this order*:
1. validator V exits from beacon chain; ETH hits pod
2. pod owner starts checkpoint
3. pod owner verifies V's withdrawal credentials using a timestamp after the checkpoint starts, but still in the same epoch

... then V's effective balance will not reflect the exit and they'll get double counted (ETH in pod counted during checkpoint + effective balance counted in verifyWC).

To prevent this, we require that `verifyWC` both:
* uses a timestamp after the current/last finalized checkpoints
* is not callable if the validator has initiated an exit